environment.py

```diff
@@ -350,4 +350,12 @@ def env() -> Mapping[str, Optional[str]]:
 
         # This is set only during integration tests. Never override this to 1.
         'AZUL_TEST_MODE': '0',
+
+        # Identifies the Terra Data Repository dataset or snapshot to index.
+        # The syntax in EBNF is:
+        # 'tdr:', Google Cloud project name, ':', ( 'dataset' | 'snapshot' ), '/', 'TDR dataset or snaphot name'
+        # Examples:
+        # tdr:broad-jade-dev-data:dataset/hca_mvp
+        # tdr:broad-jade-dev-data:snapshot/hca_mvp
```

Comment:
Unfortunately the names of the TDR datasets returned by the TDR REST API do not match the names of the corresponding BigQuery datasets. `datarepo_hca_mvp` is just listed as `hca_mvp`. I cannot find any way to recover the `datarepo_` prefix programatically. 

Here is the endpoint for listing datasets, filtered by the text "mvp":

![Screenshot from 2020-06-30 15-41-43](https://user-images.githubusercontent.com/48401926/86184226-91d42580-bae8-11ea-83cc-ff957ba60368.png)

And here is the endpoint for retrieving more detailed information on a dataset. It's cut off but the remainder of the response only describes the table schemas and provides the creation date. Note that we _could_ retrieve the Google Cloud project for a dataset/snapshot* from this endpoint instead of using the ARN syntax, if you'd prefer.

![Screenshot from 2020-06-30 15-43-34](https://user-images.githubusercontent.com/48401926/86184242-9d275100-bae8-11ea-8365-e97b72f11bab.png)

*separate endpoint for snapshots but the response is very similar

Reply:
(this limitation was part of my motivation for initially using the BigQuery nomenclature instead of Terra, so that the `datarepo_` prefix would be part of the environment configuration rather than the resolution logic).

Reply:
From slack:
Noah: "@dan Is the "convention" of using the datarepo_ prefix recoverable from any part of the TDR REST API, or perhaps another source? @Hannes has concerns about hardcoding this into this configuration logic."

Dan: "Not that I’m aware of. datrepo_ prefixes should only apply at the dataset level, so the most common flow of indexing snapshots shouldn’t need to worry about it"

Hannes: "I guess it's ok to hard-code because the worst that can happen is that we'll fail should the prefix ever change."